src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -381,7 +399,20 @@ export default class Pilot {
      * @param {Number} initialAltitude  the altitude aircraft can automatically climb to at this airport
      * @return {Array}                  [success of operation, readback]
      */
-    clearedAsFiled() {
+    clearedAsFiled(aircraft) {
+        const fms = aircraft.fms;
+        const departureRunwayModel = fms.departureRunwayModel;
```

Comment:
`fms` is unnecessary here.  instead try:
```js
const { departureRunwayModel } = fms;
```

This also seems overly defensive.  We're checking that the `departureRunwayModel` that we've already set is valid?  Shouldn't that happen when we're setting it initially and not here?  This method should really just be like: _'Ok, we have your flight plan, we verified it on instantiation, go ahead and fly'_

Reply:
I'm all for defensive programming, it just feels unnecessary here.

Reply:
Here's the scenario this protects against:
1. Aircraft is filed for SID valid only from Runway 36.
1. Aircraft is successfully  cleared as filed.
1. Aircraft is taxiied to Runway 18 (without complaints, expecting to get a new SID assigned), and the pilot silently cancels their own departure clearance because they cannot accept that rwy/sid combination.
1. Aircraft will refuse takeoff clearance until receiving a routing valid for their assigned runway, or after taxiied to a valid runway and cleared-as-filed.

So this is the portion where the pilot verifies their filed route is valid for the runway they've been assigned, and if it isn't, he will reject the route, prompting the controller to assign the appropriate SID instead.